### PR TITLE
GG-25500 Remove multiple types from CacheSqlMetadata.

### PR DESCRIPTION
**Pull request by smoldachev**

### CODE REVIEW

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java [hunk 1]
```diff
@@ -158,15 +158,15 @@ public void shouldSendCacheMetadataOnAlterTableAndCreateIndex() throws Exception
             List<CacheSqlMetadata> metadata =
                 interceptor.getListPayload(buildClusterCachesSqlMetaDest(cluster.id()), CacheSqlMetadata.class);
 
-            if (metadata == null)
+            if (metadata.isEmpty())
```
Comment (akuznetsov-gridgain) [inline modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Lets use `F.isEmpty`

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java [hunk 2]
```diff
@@ -214,15 +214,15 @@ public void shouldSendCacheMetadataOnAddAndDropdIndex() throws Exception {
...
```
Comment (akuznetsov-gridgain) [inline modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Lets use `F.isEmpty`

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java [hunk 3]
```diff
@@ -180,15 +180,15 @@ public void shouldSendCacheMetadataOnAlterTableAndCreateIndex() throws Exception
...
```
Comment (akuznetsov-gridgain) [inline modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Lets use `F.isEmpty`

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java [hunk 4]
```diff
@@ -236,15 +236,15 @@ public void shouldSendCacheMetadataOnAddAndDropdIndex() throws Exception {
...
```
Comment (akuznetsov-gridgain) [inline modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Lets use `F.isEmpty`

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java [hunk 5]
```diff
@@ -258,15 +258,15 @@ public void shouldSendCacheMetadataOnAddAndDropdIndex() throws Exception {
...
```
Comment (akuznetsov-gridgain) [inline modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Lets use `F.isEmpty`

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java (hunk 1 above)
Comment (smoldachev) [reply modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Done.

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java (hunk 2 above)
Comment (smoldachev) [reply modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Done.

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java (hunk 3 above)
Comment (smoldachev) [reply modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Done.

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java (hunk 4 above)
Comment (smoldachev) [reply modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Done.

File: modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java (hunk 5 above)
Comment (smoldachev) [reply modules/gmc-agent/src/test/java/org/apache/ignite/agent/service/CacheServiceSelfTest.java]: Done.

**Pull request merged**